src/phoenix/experimental/evals/models/openai.py

```diff
@@ -90,33 +118,46 @@ def _init_environment(self) -> None:
             )
 
     def _init_open_ai(self) -> None:
+        # For Azure, you need to provide the azure_endpoint
+        self._is_azure = self.azure_endpoint is not None
+
         self._model_uses_legacy_completion_api = self.model_name.startswith(
             LEGACY_COMPLETION_API_MODELS
         )
-        if self.openai_api_key is None:
+        if self.api_key is None:
             api_key = os.getenv(OPENAI_API_KEY_ENVVAR_NAME)
             if api_key is None:
                 # TODO: Create custom AuthenticationError
                 raise RuntimeError(
-                    "OpenAI's API key not provided. Pass it as an argument to 'openai_api_key' "
+                    "OpenAI's API key not provided. Pass it as an argument to 'api_key' "
                     "or set it in your environment: 'export OPENAI_API_KEY=sk-****'"
                 )
-            self.openai_api_key = api_key
-        self._client = self._openai.Client(api_key=self.openai_api_key)
-        self.openai_api_type = self.openai_api_type or self._openai.api_type
-        self.openai_api_version = self.openai_api_version or self._openai.api_version
-        self.openai_organization = self.openai_organization or self._openai.organization
-
-        self._is_azure = (self._openai.api_type or "").lower().startswith("azure")
+            self.api_key = api_key
 
+        self.api_version = self.api_version or self._openai.api_version
+        self.organization = self.organization or self._openai.organization
+        # Initialize specific clients depending on the API backend
+        # Set the type first
+        self._client: Union[self._openai.OpenAI, self._openai.AzureOpenAI]  # type: ignore
         if self._is_azure:
-            if not self.engine:
-                raise ValueError(
-                    "You must provide the deployment name in the 'engine' parameter "
-                    "to access the Azure OpenAI service"
-                )
-            self._openai_api_model_name = self.engine
-        elif self.model_name in MODEL_TOKEN_LIMIT_MAPPING.keys():
+            # Validate the azure options and construct a client
+            azure_options = self._get_azure_options()
+            self._client = self._openai.AzureOpenAI(
+                azure_endpoint=azure_options.azure_endpoint,
+                azure_deployment=azure_options.azure_deployment,
+                api_version=azure_options.api_version,
+                azure_ad_token=azure_options.azure_ad_token,
+                azure_ad_token_provider=azure_options.azure_ad_token_provider,
+                api_key=self.api_key,
+                organization=self.organization,
+            )
+        else:
+            self._client = self._openai.OpenAI(
+                api_key=self.api_key,
+                organization=self.organization,
```

Comment:
See my other comment about `endpoint`.
```suggestion
                organization=self.organization,
                base_url=self.endpoint,
```

Reply:
I just exposed base_url